arch/risc-v/src/mpfs/mpfs_irq.c

```diff
@@ -184,22 +149,11 @@ void up_disable_irq(int irq)
 
       /* Clear enable bit for the irq */
 
-      uint64_t hart_id = READ_CSR(mhartid);
-      uintptr_t miebase;
-
-      if (hart_id == 0)
-        {
-          miebase =  MPFS_PLIC_H0_MIE0;
-        }
-      else
-        {
-          miebase = MPFS_PLIC_H1_MIE0 +
-            ((hart_id - 1) * MPFS_HART_MIE_OFFSET);
-        }
+      uintptr_t iebase = mpfs_plic_get_iebase();
 
       if (0 <= extirq && extirq <= NR_IRQS - MPFS_IRQ_EXT_START)
         {
-          modifyreg32(miebase + (4 * (extirq / 32)), 1 << (extirq % 32), 0);
+          modifyreg32(iebase + (4 * (extirq / 32)), 1 << (extirq % 32), 0);
```

Comment:
```suggestion
          modifyreg32(iebase + (extirq / 32), 1 << (extirq % 32), 0);
```
together with `uint32_t *iebase = (uint32_t *)mpfs_plic_get_iebase();`

Or we can change `up_irqinitialize` code to:
```
  uintptr_t iebase = mpfs_plic_get_iebase();

  putreg32(0x0, iebase + 0);
  putreg32(0x0, iebase + 4);
  putreg32(0x0, iebase + 8);
  putreg32(0x0, iebase + 12);
  putreg32(0x0, iebase + 16);
  putreg32(0x0, iebase + 20);
```
Just to follow same style

Reply:
I'll modify the init function